src/Chapter04.Tests/Listing04.07.UnexpectedInequality.Tests.cs

```diff
@@ -9,23 +9,23 @@ public class ProgramTests
         [TestMethod]
         public void MainTest()
         {
-            const string netCoreApp3expected =
-@"4.2 != 4.2000002861023
-4.2 != 4.20000028610229
-(float)4.2M != 4.2F
-4.20000028610229 != 4.2
-4.2F != 4.2D
-4.19999980926514 != 4.2
-4.2F != 4.2D";
+            string netCoreApp3expected =
+$@"{(decimal)4.2} != {(decimal)4.2000002861023}
+{(double)4.2} != {(double)4.20000028610229}
+(float){(float)4.2}M != {(float)4.2}F
+{(double)4.20000028610229} != {(double)4.2}
+{(float)4.2}F != {(double)4.2D}D
+{(double)4.19999980926514F} != {4.2D}
+{4.2F}F != {4.2D}D";
```

Comment:
We don't want to use casts here. So rather than `(float)4.2` we want to use `4.2F`. For doubles, you don't need any specific suffix, simply having a decimal point is enough.